urllib3/_collections.py

```diff
@@ -278,10 +268,7 @@ def __repr__(self):
     def _copy_from(self, other):
         for key in other:
             val = other.getlist(key)
-            if isinstance(val, list):
-                # Don't need to convert tuples
-                val = list(val)
-            self._container[key.lower()] = [key] + val
+            self._container[key.lower()] = [key] + list(val)
```

Comment:
So I would rather be defensive here. The other object *might not* be a HTTPHeaderDict and could potentially not return a list. If it returns, for example, a string then we'll be doing:

```
>>> list('str')
['s', 't', 'r']
```